src/app/shared/services/auth.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { UserInterface } from '@shared/interfaces/user-interface';
+import { AngularFireAuth } from '@angular/fire/auth';
+import firebase from 'firebase';
+import UserCredential = firebase.auth.UserCredential;
+import AuthProvider = firebase.auth.AuthProvider;
+
+@Injectable()
+export class AuthService {
+    public user: UserInterface;
+    public redirectUrl: string;
+
+    constructor(public afAuth: AngularFireAuth, private router: Router) {
+    }
+
+    public googleAuth(): Promise<void> {
+        return this.authLogin(new firebase.auth.GoogleAuthProvider());
+    }
+
+    public facebookAuth(): Promise<void> {
+        return this.authLogin(new firebase.auth.FacebookAuthProvider());
+    }
+
+    public githubAuth(): Promise<void> {
+        return this.authLogin(new firebase.auth.GithubAuthProvider());
+    }
+
+    private authLogin(provider: AuthProvider): Promise<void> {
+        return this.afAuth.signInWithPopup(provider)
+            .then((result: UserCredential) => {
+                const credential = result.credential as firebase.auth.OAuthCredential;
+                const { displayName, email, photoURL, refreshToken } = result.user;
+                const { accessToken } = credential;
+                this.user = { displayName, email, photoURL, refreshToken, accessToken };
```

Comment:
Simplify it
```ts
const credential = result.credential as firebase.auth.OAuthCredential;
const { displayName, email, photoURL, refreshToken } = result.user;
const { accessToken } = credential;
this.user = { displayName, email, photoURL, refreshToken, accessToken };
```
to this:
```ts
this.user = {...result.user, ...credential.accessToken}
```

Reply:
This "result.user" has a lot of properties that I don't need
![image](https://user-images.githubusercontent.com/53089462/102980503-57d06300-4510-11eb-93b3-1608e288bad4.png)

Reply:
Yes, but if you have defined an `interface` `||` `model`, it will take only values that `key` matched with your `interface`

Reply:
Hm, I have UserInterface and  try to do the way you suggested, but I still get all properties.
![image](https://user-images.githubusercontent.com/53089462/102984266-05924080-4516-11eb-8c42-60f0565f637b.png)
![image](https://user-images.githubusercontent.com/53089462/102984414-37a3a280-4516-11eb-9f30-9b0e9b1944a0.png)
![image](https://user-images.githubusercontent.com/53089462/102984465-4b4f0900-4516-11eb-9f1e-f83576bb6f33.png)

Reply:
Yes, you are right! So leave as you wrote, that was my mistake, sorry for that! :(

Reply:
Everything is ok, thank you